public/app/features/alerting/unified/rule-list/FilterView.tsx

```diff
@@ -107,9 +114,25 @@ function FilterViewResults({ filterState }: FilterViewProps) {
   return (
     <Stack direction="column" gap={0}>
       <ul aria-label="filtered-rule-list">
-        {rules.map(({ key, rule, groupIdentifier }) => (
-          <AlertRuleLoader key={key} rule={rule} groupIdentifier={groupIdentifier} />
-        ))}
+        {rules.map((ruleWithOrigin) => {
+          const { key, rule, groupIdentifier, origin } = ruleWithOrigin;
+
+          switch (origin) {
+            case 'grafana':
+              return (
+                <GrafanaRuleLoader
+                  key={key}
+                  rule={rule}
+                  groupName={groupIdentifier.groupName}
+                  namespaceName={ruleWithOrigin.namespaceName}
+                />
+              );
+            case 'datasource':
+              return <AlertRuleLoader key={key} rule={rule} groupIdentifier={groupIdentifier} />;
```

Comment:
maybe we should rename this component now? `AlertRuleLoader` sounds more generic...